flang/test/Lower/omp-taskwait.f90

```diff
@@ -0,0 +1,24 @@
+! This test checks lowering of OpenMP taskwait Directive.
+
+! RUN: bbc -fopenmp -emit-fir %s -o - | \
+! RUN:   FileCheck %s --check-prefix=FIRDialect
+! RUN: bbc -fopenmp -emit-llvm %s -o - | \
+! RUN:   FileCheck %s --check-prefix=LLVMIRDialect
+! RUN: bbc -fopenmp -emit-fir %s -o - | \
+! RUN:   tco | FileCheck %s --check-prefix=LLVMIR
+
+program taskwait
+
+        integer :: a,b,c
+
+!$OMP TASKWAIT
+!FIRDialect: omp.taskwait
+!LLVMIRDialect: omp.taskwait
+!LLVMIR: %{{.*}} = call i32 @__kmpc_omp_taskwait(%struct.ident_t* @1, i32 %omp_global_thread_num)
```

Comment:
Is it better to be less specific about variable names (@1, omp_global_thread_num) here?

Reply:
Sure, Thanks for pointing this out.
I'll do it barrier test too. :)